src/test/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProviderTest.java

```diff
@@ -0,0 +1,113 @@
+    @Test
+    public  void test_getCredentials_not_empty(){
+        Mockito.when(credentials.getToken()).thenReturn("test-token");
+        BaseAwsCredentialsProvider baseAwsCredentialsProvider = new BaseAwsCredentialsProvider("http://testurl","test-value") {
+            @Override
+            protected void authenticate() {
+
+            }
+        };
+        baseAwsCredentialsProvider.credentials = credentials;
+
+        CerberusCredentials credentials = baseAwsCredentialsProvider.getCredentials();
+        Assert.assertNotNull(credentials);
+        assertThat(credentials.getToken()).isEqualTo("test-token");
+    }
+
+    @Test
+    public  void test_executeRequestWithRetry() throws  Exception{
+        Mockito.when(httpClient.newCall(Mockito.any())).thenReturn(call);
+        Request request = new Request.Builder().url("http://testurl").build();
+
+        Response response = new Response.Builder().request(request).protocol(Protocol.HTTP_1_0).code(200).message("response").build();
+        Mockito.when(call.execute()).thenReturn(response);
+        BaseAwsCredentialsProvider baseAwsCredentialsProvider = new BaseAwsCredentialsProvider("http://testurl",httpClient) {
+            @Override
+            protected void authenticate() {
+
+            }
+        };
+        response =  baseAwsCredentialsProvider.executeRequestWithRetry(null, 1,1);
+        Assert.assertNotNull(response);
+    }
+
+    @Test(expected = IOException.class)
+    public  void test_executeRequestWithRetry_exception() throws  Exception{
+        Mockito.when(httpClient.newCall(Mockito.any())).thenReturn(call);
+        Request request = new Request.Builder().url("http://testurl").build();
+
+        Response response = new Response.Builder().request(request).protocol(Protocol.HTTP_1_0).code(200).message("response").build();
+        Mockito.doThrow(new IOException()).when(call).execute();
+        BaseAwsCredentialsProvider baseAwsCredentialsProvider = new BaseAwsCredentialsProvider("http://testurl",httpClient) {
+            @Override
+            protected void authenticate() {
+
+            }
+        };
+        baseAwsCredentialsProvider.executeRequestWithRetry(request, 1,1);
+
+    }
+    @Test(expected = CerberusServerException.class)
+    public  void test_parseAndThrowErrorResponse() throws  Exception{
+        Mockito.when(httpClient.newCall(Mockito.any())).thenReturn(call);
+        Request request = new Request.Builder().url("http://testurl").build();
+
+        Response response = new Response.Builder().request(request).protocol(Protocol.HTTP_1_0).code(200).message("response").build();
+        Mockito.doThrow(new IOException()).when(call).execute();
+        BaseAwsCredentialsProvider baseAwsCredentialsProvider = new BaseAwsCredentialsProvider("http://testurl",httpClient) {
+            @Override
+            protected void authenticate() {
+
+            }
+        };
+        baseAwsCredentialsProvider.parseAndThrowErrorResponse(200,"response");
```

Comment:
Does this call throw a `CerberusServerException`? Because an `IOException` is thrown on line 103.

Reply:
anyhow parseAndThrowErrorResponse   method always throw the CerberusServerException , that is why we have expected that exception, IO Exception is not needed , i will remove that and push this change in next commit